crates/router/src/types/transformers.rs

```diff
@@ -96,6 +96,45 @@ impl ForeignFrom<storage_enums::AttemptStatus> for storage_enums::IntentStatus {
     }
 }
 
+impl ForeignTryFrom<storage_enums::AttemptStatus> for storage_enums::CaptureStatus {
+    type Error = error_stack::Report<errors::ApiErrorResponse>;
+
+    fn foreign_try_from(
+        attempt_status: storage_enums::AttemptStatus,
+    ) -> errors::RouterResult<Self> {
+        match attempt_status {
+            storage_enums::AttemptStatus::Charged
+            | storage_enums::AttemptStatus::PartialCharged => Ok(Self::Charged),
+            storage_enums::AttemptStatus::Pending
+            | storage_enums::AttemptStatus::CaptureInitiated => Ok(Self::Pending),
+            storage_enums::AttemptStatus::Failure
+            | storage_enums::AttemptStatus::CaptureFailed => Ok(Self::Failed),
+
+            storage_enums::AttemptStatus::Started
+            | storage_enums::AttemptStatus::AuthenticationFailed
+            | storage_enums::AttemptStatus::RouterDeclined
+            | storage_enums::AttemptStatus::AuthenticationPending
+            | storage_enums::AttemptStatus::AuthenticationSuccessful
+            | storage_enums::AttemptStatus::Authorized
+            | storage_enums::AttemptStatus::AuthorizationFailed
+            | storage_enums::AttemptStatus::Authorizing
+            | storage_enums::AttemptStatus::CodInitiated
+            | storage_enums::AttemptStatus::Voided
+            | storage_enums::AttemptStatus::VoidInitiated
+            | storage_enums::AttemptStatus::VoidFailed
+            | storage_enums::AttemptStatus::AutoRefunded
+            | storage_enums::AttemptStatus::Unresolved
+            | storage_enums::AttemptStatus::PaymentMethodAwaited
+            | storage_enums::AttemptStatus::ConfirmationAwaited
+            | storage_enums::AttemptStatus::DeviceDataCollectionPending => {
+                Err(errors::ApiErrorResponse::NotSupported {
```

Comment:
use this error enum
https://github.com/juspay/hyperswitch/blob/95810dd6d82632fc7777112c6b10edb96897af0b/crates/router/src/compatibility/stripe/errors.rs#L152